ui-charts/src/manchette/hooks/useManchetteWithSpaceTimeChart.ts

```diff
@@ -362,161 +334,361 @@ const useManchetteWithSpaceTimeChart = ({
+                const xEnd = clamp(position.x, 0, width);
+                const yStart = clamp(initialPosition.y, 0, canvasDrawingHeight);
+                const yEnd = clamp(position.y, 0, canvasDrawingHeight);
+                newPixelRect = { xStart, xEnd, yStart, yEnd };
+              }
+
+              return {
+                ...prev,
+                rect: newRect,
+                pixelRect: newPixelRect,
+              };
+            }
 
-          if (!panning) {
-            return {
-              ...prev,
-              panning: { initialOffset: { x: xOffset, y: yOffset } },
-            };
-          }
+            if (!panning) {
+              return {
+                ...prev,
+                panning: { initialOffset: { x: xOffset, y: yOffset } },
+              };
+            }
 
-          const newState = { ...prev };
-          const { initialOffset } = panning;
-          newState.xOffset = initialOffset.x + diff.x;
-
-          const newYPos = initialOffset.y - diff.y;
-          if (
-            manchetteWithSpaceTimeChartRef.current &&
-            newYPos >= 0 &&
-            newYPos + manchetteWithSpaceTimeChartRef.current.offsetHeight <
-              manchetteWithSpaceTimeChartRef.current.scrollHeight
-          ) {
-            newState.yOffset = newYPos;
-            manchetteWithSpaceTimeChartRef.current.scrollTop = newYPos;
-          }
-          return newState;
-        });
+            const newState = { ...prev };
+            const { initialOffset } = panning;
+            newState.xOffset = initialOffset.x + diff.x;
+
+            const newYPos = initialOffset.y - diff.y;
+            if (
+              manchetteWithSpaceTimeChartRef.current &&
+              newYPos >= 0 &&
+              newYPos + manchetteWithSpaceTimeChartRef.current.offsetHeight <
+                manchetteWithSpaceTimeChartRef.current.scrollHeight
+            ) {
+              newState.yOffset = newYPos;
+              manchetteWithSpaceTimeChartRef.current.scrollTop = newYPos;
+            }
+            return newState;
+          });
+        },
       },
-    }),
-    [
-      simplifiedWaypoints,
-      computedScales,
-      xZoom,
-      paths,
-      xOffset,
-      isShiftPressed,
-      state,
-      panning,
-      yOffset,
-      manchetteWithSpaceTimeChartRef,
-      handleXZoom,
-      isProportional,
-      rect,
-      spaceOrigin,
-      timeOrigin,
-      canvasDrawingHeight,
-    ]
-  );
-
-  const timeScale = useMemo(() => zoomValueToTimeScale(xZoom), [xZoom]);
-  const spaceScale = useMemo(
-    () => zoomValueToSpaceScale(minZoomMillimeterPerPx, maxZoomMillimeterPerPx, yZoom),
-    [yZoom, minZoomMillimeterPerPx, maxZoomMillimeterPerPx]
-  );
-  return useMemo(
-    () => ({
-      manchetteProps,
-      spaceTimeChartProps,
+      spaceTimeChartChildren: splitPoints.map((sp, i) =>
+        createElement(sp.spaceTimeChartComponent, {
+          key: i,
```

Comment:
Would be nicer to add an `id`/`key` prop to `SplitPoint` to avoid using indexes as keys.

Reply:
Hum, I don't see why it would be better to force users to add artificial IDs, only for some technical issue within `useManchetteWithSpaceTimeChart`. Especially since the `"react/no-array-index-key"` ESLint rule [is disabled](https://github.com/OpenRailAssociation/osrd-ui/blob/502987e0c2f596c2b34020d1309140123a760a24/.eslintrc#L79).

Reply:
I don't think the IDs would be artificial. They would e.g. be OP IDs and such.

The rule is disabled, but that doesn't mean that using indexes as keys is a good practice.
See https://react.dev/learn/rendering-lists#why-does-react-need-keys

Reply:
Fair enough, I'll add IDs.